File: functions/Hoelder_my.py
"""
Показатель Гёльдера
"""
import numpy as np
from .Hoelder_f import Hoelder_f
from .exp_smooth_twice import exp_smooth_twice
from utils import setup_figure, finalize_figure, iter_windows
from config import config


def Hoelder_my(Y, num_bands, start_offset, deviation, window_shift,
               window_width, num_windows, fig=None):
    Y = np.array(Y).flatten()
    s = config.plots.style
    alpha = config.smoothing.hoelder

    g = []
    dd = []

    for y in iter_windows(Y, start_offset, window_shift, window_width, num_windows):
        g.append(Hoelder_f(y))
        dd.append(np.max(y) - np.min(y))

    g = np.array(g)
    g_sm = exp_smooth_twice(np.abs(g).reshape(-1, 1), alpha).flatten()

    fig, [ax], standalone = setup_figure(fig, figsize=(12, 6))

    ax.plot(np.abs(g), linewidth=s.main_linewidth, label='Показатель Гёльдера')
    ax.plot(g_sm, 'k', linewidth=s.thick_linewidth, label='Сглаживание')
    ax.grid(True)
    ax.set_title('Показатель Гёльдера',
                 fontsize=s.title_fontsize, fontweight=s.title_fontweight)
    ax.set_xlabel('Номер окна', fontsize=s.small_label_fontsize)
    ax.legend()

    finalize_figure(fig, standalone)

File: matlab/Hoelder_my.m
function Hoelder_my(Y,m,K,d,D,L,R);
for k=1:R;
    y=Y(K+(k-1)*d+1:K+(k-1)*d+L);
    g(k)=Hoelder_f(y); 
    dd(k)=max(y)-min(y);
end;

hF8=figure; 
set(hF8,'Units','Normalized','Pos',[0.2 0.175 0.78 0.55],...
    'Color',[0.8 0.8 0.8]);
g_sm=exp_smooth_twice(abs(g'),0.2);

%subplot(2,1,1);
plot(abs(g),'LineWidth',3); grid; %axis([0 R 0 1]);
hold on; plot(g_sm,'k','LineWidth',4);
title('Показатель Гёльдера','FontSize',14,'FontWeight','Bold');

%subplot(2,1,2);
%plot(dd,'LineWidth',3); grid;
%title('Размах на скользящем окне','FontSize',14,'FontWeight','Bold');
%xlabel('Номер окна','FontSize',12); 